e2e/tests/interchain_accounts/gov_test.go

```diff
@@ -0,0 +1,153 @@
+package interchain_accounts
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	"github.com/gogo/protobuf/proto"
+	"github.com/strangelove-ventures/ibctest/v6"
+	"github.com/strangelove-ventures/ibctest/v6/chain/cosmos"
+	"github.com/strangelove-ventures/ibctest/v6/ibc"
+	"github.com/strangelove-ventures/ibctest/v6/test"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+	controllertypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/controller/types"
+	icatypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/types"
+	ibctesting "github.com/cosmos/ibc-go/v6/testing"
+)
+
+func TestInterchainAccountsGovTestSuite(t *testing.T) {
+	suite.Run(t, new(InterchainAccountsGovTestSuite))
+}
+
+type InterchainAccountsGovTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+func (s *InterchainAccountsGovTestSuite) TestInterchainAccountsGovIntegration() {
+	t := s.T()
+	ctx := context.TODO()
+
+	// setup relayers and connection-0 between two chains
+	// channel-0 is a transfer channel but it will not be used in this test case
+	relayer, _ := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+	controllerAccount := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	controllerAddress := controllerAccount.Bech32Address(chainA.Config().Bech32Prefix)
+
+	chainBAccount := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+	chainBAddress := chainBAccount.Bech32Address(chainB.Config().Bech32Prefix)
+
+	govModuleAddress, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+	s.Require().NoError(err)
+	s.Require().NotNil(govModuleAddress)
+
+	t.Run("submit proposal for MsgRegisterInterchainAccount", func(t *testing.T) {
```

Comment:
Is it possible to make a helper function similar to this [one](https://github.com/cosmos/ibc-go/blob/main/e2e/testsuite/testsuite.go#L454) or to reuse that helper function? I guess we need an `ExecuteGovProposalLegacy` and one for the v1 usage

Reply:
Great suggestion, I'll see what I can come up with here.